### DISCUSSION TITLE
Prototype 6 (PCB with normal audio jacks and DIN5 MIDI ports)

### DISCUSSION THREAD

**Discussion by corrados** [corrados/edrumulus#69] [General]
The vision of Edrumulus prototype 6 is to create a PCB where we use normal audio jacks and support a standard DIN-5 MIDI-IN/MIDI-OUT connector. To make the build process easier, we use wired components which can be soldered easily. A standard ESP32 developer board shall be used.

Here is a picture of prototype 6:
![190487851-1c602808-ae7e-44d2-b598-2fde2b1685d8](https://user-images.githubusercontent.com/46655886/190674983-812ac58b-5ee9-4fee-a85e-2a4e99e4c47b.png)

First tests showed that, unfortunately, the separate power supply for the bias resistors and additional capacitors could not reduce the number of spikes of the ADCs. Further investigations are needed.

**Comment by corrados** [comment]
There are other issues with this prototype:

- If you want to upload a new Edrumulus software version on the ESP32 board, it has to be unattached from the prototype 6 board for it to succeed.
- If the ESP32 board is attached to the prototype 6, you cannot use MIDI via USB. You can receive MIDI messages on the PC but you cannot send MIDI commands to the ESP32.

**Reply by thijstriemstra** [reply]
It's been a while I've looked at edrumulus but this looks awesome @corrados!! Did you get these issues fixed for this prototype? I would love to build one using jlcpcb and design a 3d-printed enclosure for it.

**Reply by corrados** [reply]
No, I haven't. Therefore, I cannot recommend this prototype.

But [I have ordered](https://github.com/corrados/edrumulus/discussions/72#discussioncomment-5468951) an [ESP32-S3 which has USB-OTG](https://github.com/corrados/edrumulus/discussions/38). If I get Edrumulus running on that board, that may be a better alternative for you.

**Reply by JJZ19** [reply]
Hi! Did you manage to make this work?
I don't intend to use the MIDI over USB. I just like the idea of non-SMD components. I hope it can still work using the STM32 USB.

**Reply by corrados** [reply]
No, I have not worked on Prototype 6 again. And I have no plans to do it, actually.

**Reply by JJZ19** [reply]
Too bad I already ordered the pcb before I read this thread. Thanks for the reply though.

**Comment by shepss123** [comment]
Hie sir ready appreciate you work I manage to compile and uplord the prototype to,what I wanted to know is how to select prototype 6 in the code and how to assign  the tx and rx for 5DIN MIDI OUT

**Reply by corrados** [reply]
The prototype board is detected automatically:
https://github.com/corrados/edrumulus/blob/292ed271c0ea24271ff656cde0e035acffbf9539/hardware.cpp#L193-L199

> how to assign the tx and rx for 5DIN MIDI OUT

On the board they should be routed correctly (hopefully :-) ).

**Reply by shepss123** [reply]
Oh thank you .exact what I wanted to know

**Reply by shepss123** [reply]
One more question do I have to change the baud rate to 31250 to use the DIN5?

**Reply by corrados** [reply]
That's a good point. What I can find in the code is:
https://github.com/corrados/edrumulus/blob/292ed271c0ea24271ff656cde0e035acffbf9539/edrumulus.ino#L49

which is a different rate. But I have tested the DIN5 and it worked fine for me. Maybe you need to adjust that value for your prototype...

**Reply by shepss123** [reply]
Thank you very much

**Reply by shepss123** [reply]
And the which set of RX and TX should I used for DIN5 ports?

**Reply by corrados** [reply]
These are the schematics I have used:
![grafik](https://github.com/user-attachments/assets/d0b38d57-28cf-4572-9647-082fa05ef843)
(Sorry for the "Not yet supported" text but I have used the PDF from Prototype 2)

**Reply by shepss123** [reply]
Thank you vèry much will give it a try when will tell you the results

**Reply by shepss123** [reply]
After hour of testing it finally worked with 31250 ,am so happy

**Comment by shepss123** [comment]
First test with the drumpads,I just couldn't ignore the number of ADC spikes I was getting.did my own investigation only to find out that the 220ohm I was using was wrong
![Screenshot_20250215_090740_Drive](https://github.com/user-attachments/assets/d8518485-65d6-4740-9e72-78e49c0bee42)

**Reply by corrados** [reply]
Good to know. Thanks for the info.
